### DISCUSSION TITLE
FCM vs noFCM

### DISCUSSION THREAD

**Discussion by Coool** [tateisu/SubwayTooter#269] [Q&A]
Can you explain which app end user to choose? With Firebase Cloud Messaging (FCM) or without (noFCM)? Pros and Minus?

**Comment by tateisu** [comment]
- on the Play Store, FCM variant is distributed.

- on the F-Droid(Izzy repo),  noFCM variant is distributed.

If you don't mind about non-OSS code, you should use the FCM variant. that
include default push delivery client library made by Google.  the
difference is only that.

This app allows you to select the push delivery service that used to
receive  push notifications.

FCM (Firebase Cloud Messaging) is one of them, and it can be used without
additional installation on many devices that include the "Play developer
Service" app.

However, since it is not OSS, the ST's variation distributed in the F-Droid
(Izzy repo.) repository does not include the FCM client library, that was
demanded by repository owner.

To use push notifications with noFcm variant, you need to install another
push delivery service and then select it in the ST app settings.

please see UnifiedPush https://unifiedpush.org/ to find push delivery
service.




2024年9月22日(日) 15:58 Coool ***@***.***>:

**Comment by Coool** [comment]
How do those push notifications get to the end user? Are they related to notifications of new toot messages or they something different?

**Comment by tateisu** [comment]
you can try "account setting/update push subscription".

2024年9月22日(日) 20:04 Coool ***@***.***>:

**Comment by Coool** [comment]
I'm more inclined to use noFCM, I'll install the noFCM version and play around and then write back with my impressions.
